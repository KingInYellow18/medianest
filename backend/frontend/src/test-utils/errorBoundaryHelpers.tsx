/**
 * Error Boundary Testing Helpers
 *
 * This module provides utilities to properly test React Error Boundaries
 * with Vitest, addressing the challenge where React errors are caught by
 * the testing framework before Error Boundaries can handle them.
 */

import { render, RenderOptions } from '@testing-library/react';
import React from 'react';

// Spy to track if error boundaries are working
const errorBoundaryTestSpy: any = null;

// Configure error boundary testing environment
export const configureErrorBoundaryTesting = () => {
  // Store original console.error
  const originalConsoleError = console.error;

  // Create custom error handler that doesn't interfere with Error Boundary tests
  console.error = (...args: any[]) => {
    const message = args[0];
    if (typeof message === 'string') {
      // Known error messages that should be handled by ErrorBoundary
      const errorBoundaryMessages = [
        'Test error message',
        'Stack trace test',
        'Callback test error',
        'First error',
        'Custom fallback test',
        'Fallback info test',
        'Original error',
        'Retry test error',
        'Retry functionality test',
        'Timeout test error',
        'Accessibility test',
        'Button accessibility test',
        'Keyboard test',
        'Screen reader test',
        'Deep nested error',
        'Development logging test',
        'Production logging test',
        'HOC error test',
        'Wrapped component error',
        'Error with <script>',
        'very long error message',
      ];

      // If this is an expected error boundary test error, suppress it
      if (errorBoundaryMessages.some((msg) => message.includes(msg))) {
        return;
      }

      // Also suppress React's error boundary warnings
      if (
        message.includes('Error boundaries should implement getDerivedStateFromError') ||
        message.includes('componentDidCatch') ||
        message.includes('The above error occurred in the')
      ) {
        return;
      }
    }

    // For other errors, use original console.error
    originalConsoleError(...args);
  };

  return originalConsoleError;
};

// Restore original console.error
export const restoreErrorBoundaryTesting = (originalConsoleError: any) => {
  console.error = originalConsoleError;
};

// Render function optimized for error boundary testing
export const renderWithErrorBoundarySupport = (ui: React.ReactElement, options?: RenderOptions) => {
  const originalConsoleError = configureErrorBoundaryTesting();

  try {
    return render(ui, options);
  } finally {
    // Don't restore immediately - let the test complete first
    // This allows assertions to run without console spam
    setTimeout(() => restoreErrorBoundaryTesting(originalConsoleError), 0);
  }
};

// Test helper that wraps test functions with proper error boundary setup
export const withErrorBoundarySupport = (testFn: () => void | Promise<void>) => {
  return async () => {
    const originalConsoleError = configureErrorBoundaryTesting();

    try {
      await testFn();
    } finally {
      restoreErrorBoundaryTesting(originalConsoleError);
    }
  };
};

// Error throwing component for consistent testing
export const TestErrorComponent = ({
  message = 'Test error',
  shouldThrow = true,
}: {
  message?: string;
  shouldThrow?: boolean;
}) => {
  if (shouldThrow) {
    throw new Error(message);
  }
  return <div data-testid='no-error'>No error thrown</div>;
};
