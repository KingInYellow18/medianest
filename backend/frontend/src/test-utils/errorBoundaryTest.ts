/**
 * Utility for testing React Error Boundaries with Vitest
 *
 * This provides a clean way to test error boundaries by temporarily
 * suppressing React's error logging during error boundary tests.
 */

// Store original console.error function
const originalError = console.error;

export const suppressReactErrorLogs = (callback: () => void) => {
  // Temporarily suppress console.error to prevent React error logs during tests
  console.error = () => {};

  try {
    callback();
  } finally {
    // Restore original console.error
    console.error = originalError;
  }
};

export const withErrorBoundarySuppression = (testFn: () => void | Promise<void>) => {
  return () => {
    console.error = () => {};
    try {
      return testFn();
    } finally {
      console.error = originalError;
    }
  };
};
