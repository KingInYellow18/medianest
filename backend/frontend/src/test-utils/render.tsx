/**
 * Custom Render Utilities for React Testing Library
 * Provides enhanced testing utilities with providers and helpers
 */

import { render, RenderOptions } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React, { ReactElement, ReactNode } from 'react';
import { vi } from 'vitest';

// Mock Error Boundary Provider for testing error states
export const TestErrorBoundaryProvider = ({
  children,
  shouldError = false,
}: {
  children: ReactNode;
  shouldError?: boolean;
}) => {
  if (shouldError) {
    throw new Error('Test Error Boundary Trigger');
  }
  return <>{children}</>;
};

// Custom render function with providers
const AllTheProviders = ({ children }: { children: ReactNode }) => {
  return <div data-testid='test-wrapper'>{children}</div>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) => {
  const user = userEvent.setup();

  return {
    user,
    ...render(ui, { wrapper: AllTheProviders, ...options }),
  };
};

// Component that throws an error for testing ErrorBoundary
export const ThrowError = ({
  shouldThrow = true,
  message = 'Test error message',
}: {
  shouldThrow?: boolean;
  message?: string;
}) => {
  if (shouldThrow) {
    throw new Error(message);
  }
  return <div>No error thrown</div>;
};

// Component that throws async error
export const ThrowAsyncError = ({
  shouldThrow = true,
  message = 'Async test error',
  delay = 0,
}: {
  shouldThrow?: boolean;
  message?: string;
  delay?: number;
}) => {
  React.useEffect(() => {
    if (shouldThrow) {
      setTimeout(() => {
        throw new Error(message);
      }, delay);
    }
  }, [shouldThrow, message, delay]);

  return <div>Async component</div>;
};

// Test data generators with unique IDs
let serviceCounter = 0;
export const createMockService = (overrides: Record<string, any> = {}) => {
  serviceCounter++;
  return {
    id: `test-service-${serviceCounter}` as any,
    name: 'Test Service',
    status: 'active' as const,
    lastChecked: new Date('2025-01-12T10:00:00Z'),
    uptime: 0.995,
    responseTime: 150,
    errorCount: 2,
    ...overrides,
  };
};

export const createMockServiceWithError = () =>
  createMockService({
    status: 'error',
    errorCount: 5,
    uptime: 0.85,
    responseTime: undefined,
  });

export const createMockServiceInactive = () =>
  createMockService({
    status: 'inactive',
    uptime: 0.0,
    responseTime: undefined,
    errorCount: 0,
  });

// Accessibility testing utilities
export const axeMatchers = {
  toHaveNoViolations: expect.extend({
    toHaveNoViolations(received) {
      const pass = received.violations.length === 0;
      if (pass) {
        return {
          message: () => 'Expected accessibility violations, but none were found',
          pass: true,
        };
      } else {
        return {
          message: () =>
            `Expected no accessibility violations, but found:\n${received.violations
              .map((v: any) => `- ${v.description}`)
              .join('\n')}`,
          pass: false,
        };
      }
    },
  }),
};

// Performance testing utilities
export const measureRenderTime = async (renderFn: () => void) => {
  const start = performance.now();
  renderFn();
  const end = performance.now();
  return end - start;
};

// Error boundary testing utilities
export const suppressErrorOutput = (testFn: () => void) => {
  const originalError = console.error;
  const originalWarn = console.warn;
  console.error = () => {};
  console.warn = () => {};

  try {
    testFn();
  } finally {
    console.error = originalError;
    console.warn = originalWarn;
  }
};

// Mock ErrorBoundary that doesn't interfere with testing
export const MockErrorBoundary = ({
  children,
  onError,
  fallback,
}: {
  children: React.ReactNode;
  onError?: (error: Error, errorInfo: any) => void;
  fallback?: (error: Error, errorInfo: any) => React.ReactNode;
}) => {
  return <React.Suspense fallback={<div>Loading...</div>}>{children}</React.Suspense>;
};

// Test helper that wraps components for error testing
export const renderWithErrorHandling = (ui: ReactElement, options?: RenderOptions) => {
  // Store original console methods
  const originalError = console.error;
  const originalWarn = console.warn;

  // Suppress React error logging during tests
  console.error = () => {};
  console.warn = () => {};

  try {
    return render(ui, options);
  } finally {
    // Restore console methods
    console.error = originalError;
    console.warn = originalWarn;
  }
};

// Test helper for error boundary testing with proper error catching
export const renderWithErrorBoundary = (ui: ReactElement, options?: RenderOptions) => {
  // Temporarily suppress React's error logging
  const originalConsoleError = console.error;
  console.error = () => {};

  const result = render(ui, options);

  // Restore console.error after render
  console.error = originalConsoleError;

  return result;
};

export * from '@testing-library/react';
export { customRender as render };
