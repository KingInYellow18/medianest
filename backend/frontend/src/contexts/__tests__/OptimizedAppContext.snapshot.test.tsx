/**
 * Snapshot Tests for OptimizedAppContext
 * Tests visual consistency of provider and consumer components across different states
 */

import React from 'react';
import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest';
import { render, fireEvent, act } from '../../test-utils/render';
import { 
  AppProvider, 
  useApp, 
  useAppUser, 
  useAppSession, 
  useAppUI, 
  useAppActions,
  useIsAuthenticated,
  useHasNotifications
} from '../OptimizedAppContext';

// Test components for different context states
const UserDisplay = () => {
  const user = useAppUser();
  return (
    <div data-testid="user-display">
      <div data-testid="user-id">{user.id || 'No ID'}</div>
      <div data-testid="user-email">{user.email || 'No Email'}</div>
      <div data-testid="user-name">{user.name || 'No Name'}</div>
      <div data-testid="user-role">{user.role || 'No Role'}</div>
    </div>
  );
};

const SessionDisplay = () => {
  const session = useAppSession();
  return (
    <div data-testid="session-display">
      <div data-testid="session-id">{session.id || 'No Session ID'}</div>
      <div data-testid="session-auth">{session.isAuthenticated ? 'Authenticated' : 'Not Authenticated'}</div>
      <div data-testid="session-expires">
        {session.expiresAt ? session.expiresAt.toISOString() : 'No Expiry'}
      </div>
    </div>
  );
};

const UIDisplay = () => {
  const ui = useAppUI();
  return (
    <div data-testid="ui-display">
      <div data-testid="ui-theme">{ui.theme}</div>
      <div data-testid="ui-sidebar">{ui.sidebarOpen ? 'Open' : 'Closed'}</div>
      <div data-testid="ui-notifications-count">{ui.notifications.length}</div>
      <div data-testid="ui-notifications">
        {ui.notifications.map((notification) => (
          <div key={notification.id} className="notification" data-type={notification.type}>
            <span className="message">{notification.message}</span>
            <span className="timestamp">{notification.timestamp.toISOString()}</span>
          </div>
        ))}
      </div>
    </div>
  );
};

const ActionsDisplay = () => {
  const actions = useAppActions();
  const isAuthenticated = useIsAuthenticated();
  const hasNotifications = useHasNotifications();
  
  return (
    <div data-testid="actions-display">
      <div data-testid="is-authenticated">{isAuthenticated ? 'Authenticated' : 'Not Authenticated'}</div>
      <div data-testid="has-notifications">{hasNotifications ? 'Has Notifications' : 'No Notifications'}</div>
      <div className="action-buttons">
        <button 
          data-testid="toggle-sidebar-btn" 
          onClick={actions.toggleSidebar}
        >
          Toggle Sidebar
        </button>
        <button 
          data-testid="add-notification-btn" 
          onClick={() => actions.addNotification({
            message: 'Test notification',
            type: 'info'
          })}
        >
          Add Notification
        </button>
        <button 
          data-testid="clear-notifications-btn" 
          onClick={actions.clearNotifications}
        >
          Clear Notifications
        </button>
        <button 
          data-testid="logout-btn" 
          onClick={actions.logout}
        >
          Logout
        </button>
      </div>
    </div>
  );
};

const CompleteAppDisplay = () => (
  <div data-testid="complete-app">
    <UserDisplay />
    <SessionDisplay />
    <UIDisplay />
    <ActionsDisplay />
  </div>
);

describe('OptimizedAppContext Snapshot Tests', () => {
  beforeEach(() => {
    vi.useFakeTimers();
    vi.setSystemTime(new Date('2025-01-12T12:00:00Z'));
  });

  afterEach(() => {
    vi.useRealTimers();
  });

  describe('Initial State Snapshots', () => {
    it('should match snapshot with default initial state', () => {
      const { container } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );
      
      expect(container.firstChild).toMatchSnapshot('app-context-default-state');
    });

    it('should match snapshot with custom initial state', () => {
      const initialState = {
        user: {
          id: 'user-123' as any,
          email: 'test@example.com',
          name: 'Test User',
          role: 'admin' as const,
        },
        session: {
          id: 'session-456' as any,
          isAuthenticated: true,
          expiresAt: new Date('2025-01-12T18:00:00Z'),
        },
        ui: {
          theme: 'dark' as const,
          sidebarOpen: true,
          notifications: [],
        },
      };
      
      const { container } = render(
        <AppProvider initialState={initialState}>
          <CompleteAppDisplay />
        </AppProvider>
      );
      
      expect(container.firstChild).toMatchSnapshot('app-context-custom-initial-state');
    });

    it('should match snapshot with partially filled initial state', () => {
      const partialState = {
        user: {
          name: 'Partial User',
          email: 'partial@example.com',
        },
        ui: {
          theme: 'system' as const,
          sidebarOpen: false,
        },
      };
      
      const { container } = render(
        <AppProvider initialState={partialState}>
          <CompleteAppDisplay />
        </AppProvider>
      );
      
      expect(container.firstChild).toMatchSnapshot('app-context-partial-initial-state');
    });
  });

  describe('User State Snapshots', () => {
    it('should match snapshot with logged in admin user', () => {
      const { container, getByTestId } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );

      const actions = useAppActions();
      act(() => {
        actions.setUser({
          id: 'admin-001' as any,
          email: 'admin@company.com',
          name: 'System Administrator',
          role: 'admin',
        });
        actions.setSession({
          id: 'admin-session' as any,
          isAuthenticated: true,
          expiresAt: new Date('2025-01-12T20:00:00Z'),
        });
      });
      
      expect(container.firstChild).toMatchSnapshot('app-context-admin-user');
    });

    it('should match snapshot with regular user', () => {
      const { container } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );

      const actions = useAppActions();
      act(() => {
        actions.setUser({
          id: 'user-002' as any,
          email: 'user@company.com',
          name: 'Regular User',
          role: 'user',
        });
        actions.setSession({
          id: 'user-session' as any,
          isAuthenticated: true,
          expiresAt: new Date('2025-01-12T16:00:00Z'),
        });
      });
      
      expect(container.firstChild).toMatchSnapshot('app-context-regular-user');
    });

    it('should match snapshot with user having long name and email', () => {
      const { container } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );

      const actions = useAppActions();
      act(() => {
        actions.setUser({
          id: 'long-name-user' as any,
          email: 'very.long.email.address.for.testing@extremely-long-domain-name.example.com',
          name: 'This Is A Very Long User Name That Might Cause Display Issues',
          role: 'user',
        });
      });
      
      expect(container.firstChild).toMatchSnapshot('app-context-long-user-data');
    });
  });

  describe('Session State Snapshots', () => {
    it('should match snapshot with active session', () => {
      const { container } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );

      const actions = useAppActions();
      act(() => {
        actions.setSession({
          id: 'active-session-123' as any,
          isAuthenticated: true,
          expiresAt: new Date('2025-01-12T15:30:00Z'),
        });
      });
      
      expect(container.firstChild).toMatchSnapshot('app-context-active-session');
    });

    it('should match snapshot with expired session', () => {
      const { container } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );

      const actions = useAppActions();
      act(() => {
        actions.setSession({
          id: 'expired-session' as any,
          isAuthenticated: false,
          expiresAt: new Date('2025-01-12T10:00:00Z'), // Past time
        });
      });
      
      expect(container.firstChild).toMatchSnapshot('app-context-expired-session');
    });

    it('should match snapshot with no session', () => {
      const { container } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );
      
      expect(container.firstChild).toMatchSnapshot('app-context-no-session');
    });
  });

  describe('UI State Snapshots', () => {
    it('should match snapshot with dark theme and open sidebar', () => {
      const { container } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );

      const actions = useAppActions();
      act(() => {
        actions.setTheme('dark');
        actions.toggleSidebar();
      });
      
      expect(container.firstChild).toMatchSnapshot('app-context-dark-theme-sidebar-open');
    });

    it('should match snapshot with light theme and closed sidebar', () => {
      const { container } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );

      const actions = useAppActions();
      act(() => {
        actions.setTheme('light');
      });
      
      expect(container.firstChild).toMatchSnapshot('app-context-light-theme-sidebar-closed');
    });

    it('should match snapshot with system theme', () => {
      const { container } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );

      const actions = useAppActions();
      act(() => {
        actions.setTheme('system');
      });
      
      expect(container.firstChild).toMatchSnapshot('app-context-system-theme');
    });
  });

  describe('Notifications State Snapshots', () => {
    it('should match snapshot with single notification', () => {
      const { container } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );

      const actions = useAppActions();
      act(() => {
        actions.addNotification({
          message: 'Welcome to the application!',
          type: 'success'
        });
      });
      
      expect(container.firstChild).toMatchSnapshot('app-context-single-notification');
    });

    it('should match snapshot with multiple notifications of different types', () => {
      const { container } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );

      const actions = useAppActions();
      act(() => {
        actions.addNotification({
          message: 'Information: System updated',
          type: 'info'
        });
        actions.addNotification({
          message: 'Success: Data saved successfully',
          type: 'success'
        });
        actions.addNotification({
          message: 'Warning: Storage almost full',
          type: 'warning'
        });
        actions.addNotification({
          message: 'Error: Connection failed',
          type: 'error'
        });
      });
      
      expect(container.firstChild).toMatchSnapshot('app-context-multiple-notifications');
    });

    it('should match snapshot with long notification message', () => {
      const { container } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );

      const actions = useAppActions();
      act(() => {
        actions.addNotification({
          message: 'This is a very long notification message that contains a lot of information and might cause layout issues in the user interface. It includes multiple sentences and technical details that need to be displayed properly to the user without breaking the layout or causing overflow issues.',
          type: 'info'
        });
      });
      
      expect(container.firstChild).toMatchSnapshot('app-context-long-notification');
    });

    it('should match snapshot with special characters in notification', () => {
      const { container } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );

      const actions = useAppActions();
      act(() => {
        actions.addNotification({
          message: 'Special chars: <script>alert("xss")</script> & 🚀 日本語 العربية',
          type: 'warning'
        });
      });
      
      expect(container.firstChild).toMatchSnapshot('app-context-special-chars-notification');
    });
  });

  describe('Combined State Snapshots', () => {
    it('should match snapshot with complete application state', () => {
      const { container } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );

      const actions = useAppActions();
      act(() => {
        // Set user
        actions.setUser({
          id: 'complete-user' as any,
          email: 'complete@test.com',
          name: 'Complete User',
          role: 'admin',
        });

        // Set session
        actions.setSession({
          id: 'complete-session' as any,
          isAuthenticated: true,
          expiresAt: new Date('2025-01-12T18:00:00Z'),
        });

        // Set UI state
        actions.setTheme('dark');
        actions.toggleSidebar();

        // Add notifications
        actions.addNotification({
          message: 'System initialized successfully',
          type: 'success'
        });
        actions.addNotification({
          message: 'New features available',
          type: 'info'
        });
      });
      
      expect(container.firstChild).toMatchSnapshot('app-context-complete-state');
    });

    it('should match snapshot after logout action', () => {
      const { container } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );

      const actions = useAppActions();
      act(() => {
        // First set up some state
        actions.setUser({
          id: 'logout-user' as any,
          email: 'logout@test.com',
          name: 'Logout User',
          role: 'user',
        });
        actions.setSession({
          id: 'logout-session' as any,
          isAuthenticated: true,
          expiresAt: new Date('2025-01-12T18:00:00Z'),
        });
        actions.addNotification({
          message: 'You are logged in',
          type: 'info'
        });
        actions.toggleSidebar();

        // Then logout
        actions.logout();
      });
      
      expect(container.firstChild).toMatchSnapshot('app-context-after-logout');
    });
  });

  describe('Interactive Action Snapshots', () => {
    it('should match snapshot after sidebar toggle interaction', () => {
      const { container, getByTestId } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );

      const toggleButton = getByTestId('toggle-sidebar-btn');
      act(() => {
        fireEvent.click(toggleButton);
      });
      
      expect(container.firstChild).toMatchSnapshot('app-context-after-sidebar-toggle');
    });

    it('should match snapshot after adding notification via button', () => {
      const { container, getByTestId } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );

      const addButton = getByTestId('add-notification-btn');
      act(() => {
        fireEvent.click(addButton);
      });
      
      expect(container.firstChild).toMatchSnapshot('app-context-after-add-notification');
    });

    it('should match snapshot after clearing notifications', () => {
      const { container, getByTestId } = render(
        <AppProvider>
          <CompleteAppDisplay />
        </AppProvider>
      );

      const addButton = getByTestId('add-notification-btn');
      const clearButton = getByTestId('clear-notifications-btn');
      
      act(() => {
        fireEvent.click(addButton);
        fireEvent.click(addButton);
        fireEvent.click(clearButton);
      });
      
      expect(container.firstChild).toMatchSnapshot('app-context-after-clear-notifications');
    });
  });

  describe('Individual Hook Snapshots', () => {
    it('should match snapshot of user data component only', () => {
      const { container } = render(
        <AppProvider initialState={{
          user: {
            id: 'isolated-user' as any,
            email: 'isolated@test.com',
            name: 'Isolated User',
            role: 'user',
          }
        }}>
          <UserDisplay />
        </AppProvider>
      );
      
      expect(container.firstChild).toMatchSnapshot('app-context-user-component-only');
    });

    it('should match snapshot of session component only', () => {
      const { container } = render(
        <AppProvider initialState={{
          session: {
            id: 'isolated-session' as any,
            isAuthenticated: true,
            expiresAt: new Date('2025-01-12T20:00:00Z'),
          }
        }}>
          <SessionDisplay />
        </AppProvider>
      );
      
      expect(container.firstChild).toMatchSnapshot('app-context-session-component-only');
    });

    it('should match snapshot of UI component only', () => {
      const { container } = render(
        <AppProvider initialState={{
          ui: {
            theme: 'dark' as const,
            sidebarOpen: true,
            notifications: [
              {
                id: 'test-1',
                message: 'Test notification',
                type: 'info' as const,
                timestamp: new Date('2025-01-12T12:00:00Z'),
              }
            ],
          }
        }}>
          <UIDisplay />
        </AppProvider>
      );
      
      expect(container.firstChild).toMatchSnapshot('app-context-ui-component-only');
    });
  });

  describe('Error Boundary Integration Snapshots', () => {
    it('should match snapshot when provider is missing', () => {
      // Test component that tries to use context without provider
      const OrphanComponent = () => {
        try {
          const { state } = useApp();
          return <div>Should not render</div>;
        } catch (error) {
          return (
            <div data-testid="context-error">
              Error: {(error as Error).message}
            </div>
          );
        }
      };

      const { container } = render(<OrphanComponent />);
      
      expect(container.firstChild).toMatchSnapshot('app-context-missing-provider-error');
    });
  });
});