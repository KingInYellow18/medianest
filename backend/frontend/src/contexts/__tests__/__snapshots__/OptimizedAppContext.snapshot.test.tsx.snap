// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OptimizedAppContext Snapshot Tests > Error Boundary Integration Snapshots > should match snapshot when provider is missing > app-context-missing-provider-error 1`] = `
<div
  data-testid="test-wrapper"
>
  <div
    data-testid="context-error"
  >
    Error: 
    useApp must be used within an AppProvider
  </div>
</div>
`;

exports[`OptimizedAppContext Snapshot Tests > Individual Hook Snapshots > should match snapshot of UI component only > app-context-session-component-only 1`] = `
<div
  data-testid="test-wrapper"
>
  <div
    data-testid="session-display"
  >
    <div
      data-testid="session-id"
    >
      isolated-session
    </div>
    <div
      data-testid="session-auth"
    >
      Authenticated
    </div>
    <div
      data-testid="session-expires"
    >
      2025-01-12T20:00:00.000Z
    </div>
  </div>
</div>
`;

exports[`OptimizedAppContext Snapshot Tests > Individual Hook Snapshots > should match snapshot of UI component only > app-context-ui-component-only 1`] = `
<div
  data-testid="test-wrapper"
>
  <div
    data-testid="ui-display"
  >
    <div
      data-testid="ui-theme"
    >
      dark
    </div>
    <div
      data-testid="ui-sidebar"
    >
      Open
    </div>
    <div
      data-testid="ui-notifications-count"
    >
      1
    </div>
    <div
      data-testid="ui-notifications"
    >
      <div
        class="notification"
        data-type="info"
      >
        <span
          class="message"
        >
          Test notification
        </span>
        <span
          class="timestamp"
        >
          2025-01-12T12:00:00.000Z
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`OptimizedAppContext Snapshot Tests > Individual Hook Snapshots > should match snapshot of UI component only > app-context-user-component-only 1`] = `
<div
  data-testid="test-wrapper"
>
  <div
    data-testid="user-display"
  >
    <div
      data-testid="user-id"
    >
      isolated-user
    </div>
    <div
      data-testid="user-email"
    >
      isolated@test.com
    </div>
    <div
      data-testid="user-name"
    >
      Isolated User
    </div>
    <div
      data-testid="user-role"
    >
      user
    </div>
  </div>
</div>
`;

exports[`OptimizedAppContext Snapshot Tests > Initial State Snapshots > should match snapshot with partially filled initial state > app-context-custom-initial-state 1`] = `
<div
  data-testid="test-wrapper"
>
  <div
    data-testid="complete-app"
  >
    <div
      data-testid="user-display"
    >
      <div
        data-testid="user-id"
      >
        user-123
      </div>
      <div
        data-testid="user-email"
      >
        test@example.com
      </div>
      <div
        data-testid="user-name"
      >
        Test User
      </div>
      <div
        data-testid="user-role"
      >
        admin
      </div>
    </div>
    <div
      data-testid="session-display"
    >
      <div
        data-testid="session-id"
      >
        session-456
      </div>
      <div
        data-testid="session-auth"
      >
        Authenticated
      </div>
      <div
        data-testid="session-expires"
      >
        2025-01-12T18:00:00.000Z
      </div>
    </div>
    <div
      data-testid="ui-display"
    >
      <div
        data-testid="ui-theme"
      >
        dark
      </div>
      <div
        data-testid="ui-sidebar"
      >
        Open
      </div>
      <div
        data-testid="ui-notifications-count"
      >
        0
      </div>
      <div
        data-testid="ui-notifications"
      />
    </div>
    <div
      data-testid="actions-display"
    >
      <div
        data-testid="is-authenticated"
      >
        Authenticated
      </div>
      <div
        data-testid="has-notifications"
      >
        No Notifications
      </div>
      <div
        class="action-buttons"
      >
        <button
          data-testid="toggle-sidebar-btn"
        >
          Toggle Sidebar
        </button>
        <button
          data-testid="add-notification-btn"
        >
          Add Notification
        </button>
        <button
          data-testid="clear-notifications-btn"
        >
          Clear Notifications
        </button>
        <button
          data-testid="logout-btn"
        >
          Logout
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`OptimizedAppContext Snapshot Tests > Initial State Snapshots > should match snapshot with partially filled initial state > app-context-default-state 1`] = `
<div
  data-testid="test-wrapper"
>
  <div
    data-testid="complete-app"
  >
    <div
      data-testid="user-display"
    >
      <div
        data-testid="user-id"
      >
        No ID
      </div>
      <div
        data-testid="user-email"
      >
        No Email
      </div>
      <div
        data-testid="user-name"
      >
        No Name
      </div>
      <div
        data-testid="user-role"
      >
        No Role
      </div>
    </div>
    <div
      data-testid="session-display"
    >
      <div
        data-testid="session-id"
      >
        No Session ID
      </div>
      <div
        data-testid="session-auth"
      >
        Not Authenticated
      </div>
      <div
        data-testid="session-expires"
      >
        No Expiry
      </div>
    </div>
    <div
      data-testid="ui-display"
    >
      <div
        data-testid="ui-theme"
      >
        system
      </div>
      <div
        data-testid="ui-sidebar"
      >
        Closed
      </div>
      <div
        data-testid="ui-notifications-count"
      >
        0
      </div>
      <div
        data-testid="ui-notifications"
      />
    </div>
    <div
      data-testid="actions-display"
    >
      <div
        data-testid="is-authenticated"
      >
        Not Authenticated
      </div>
      <div
        data-testid="has-notifications"
      >
        No Notifications
      </div>
      <div
        class="action-buttons"
      >
        <button
          data-testid="toggle-sidebar-btn"
        >
          Toggle Sidebar
        </button>
        <button
          data-testid="add-notification-btn"
        >
          Add Notification
        </button>
        <button
          data-testid="clear-notifications-btn"
        >
          Clear Notifications
        </button>
        <button
          data-testid="logout-btn"
        >
          Logout
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`OptimizedAppContext Snapshot Tests > Interactive Action Snapshots > should match snapshot after clearing notifications > app-context-after-sidebar-toggle 1`] = `
<div
  data-testid="test-wrapper"
>
  <div
    data-testid="complete-app"
  >
    <div
      data-testid="user-display"
    >
      <div
        data-testid="user-id"
      >
        No ID
      </div>
      <div
        data-testid="user-email"
      >
        No Email
      </div>
      <div
        data-testid="user-name"
      >
        No Name
      </div>
      <div
        data-testid="user-role"
      >
        No Role
      </div>
    </div>
    <div
      data-testid="session-display"
    >
      <div
        data-testid="session-id"
      >
        No Session ID
      </div>
      <div
        data-testid="session-auth"
      >
        Not Authenticated
      </div>
      <div
        data-testid="session-expires"
      >
        No Expiry
      </div>
    </div>
    <div
      data-testid="ui-display"
    >
      <div
        data-testid="ui-theme"
      >
        system
      </div>
      <div
        data-testid="ui-sidebar"
      >
        Open
      </div>
      <div
        data-testid="ui-notifications-count"
      >
        0
      </div>
      <div
        data-testid="ui-notifications"
      />
    </div>
    <div
      data-testid="actions-display"
    >
      <div
        data-testid="is-authenticated"
      >
        Not Authenticated
      </div>
      <div
        data-testid="has-notifications"
      >
        No Notifications
      </div>
      <div
        class="action-buttons"
      >
        <button
          data-testid="toggle-sidebar-btn"
        >
          Toggle Sidebar
        </button>
        <button
          data-testid="add-notification-btn"
        >
          Add Notification
        </button>
        <button
          data-testid="clear-notifications-btn"
        >
          Clear Notifications
        </button>
        <button
          data-testid="logout-btn"
        >
          Logout
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`OptimizedAppContext Snapshot Tests > Session State Snapshots > should match snapshot with no session > app-context-no-session 1`] = `
<div
  data-testid="test-wrapper"
>
  <div
    data-testid="complete-app"
  >
    <div
      data-testid="user-display"
    >
      <div
        data-testid="user-id"
      >
        No ID
      </div>
      <div
        data-testid="user-email"
      >
        No Email
      </div>
      <div
        data-testid="user-name"
      >
        No Name
      </div>
      <div
        data-testid="user-role"
      >
        No Role
      </div>
    </div>
    <div
      data-testid="session-display"
    >
      <div
        data-testid="session-id"
      >
        No Session ID
      </div>
      <div
        data-testid="session-auth"
      >
        Not Authenticated
      </div>
      <div
        data-testid="session-expires"
      >
        No Expiry
      </div>
    </div>
    <div
      data-testid="ui-display"
    >
      <div
        data-testid="ui-theme"
      >
        system
      </div>
      <div
        data-testid="ui-sidebar"
      >
        Closed
      </div>
      <div
        data-testid="ui-notifications-count"
      >
        0
      </div>
      <div
        data-testid="ui-notifications"
      />
    </div>
    <div
      data-testid="actions-display"
    >
      <div
        data-testid="is-authenticated"
      >
        Not Authenticated
      </div>
      <div
        data-testid="has-notifications"
      >
        No Notifications
      </div>
      <div
        class="action-buttons"
      >
        <button
          data-testid="toggle-sidebar-btn"
        >
          Toggle Sidebar
        </button>
        <button
          data-testid="add-notification-btn"
        >
          Add Notification
        </button>
        <button
          data-testid="clear-notifications-btn"
        >
          Clear Notifications
        </button>
        <button
          data-testid="logout-btn"
        >
          Logout
        </button>
      </div>
    </div>
  </div>
</div>
`;
